### PR TITLE
[KED-2981] Re-write Kedro UI buttons

### PR DESCRIPTION
**Pull request by rashidakanchwala**
## Description

This PR is a part of the Kedro-UI migration ticket. In this PR we migrate the button functionality from Kedro-ui to Kedro-viz 

## Development notes

The Kedro-UI button functionality had lot of styles that we weren't using in Kedro-viz. So I have made that simple in this PR. In [Kedro-UI](https://quantumblacklabs.github.io/kedro-ui/), we have primary button, second button, small button, button with wipe transition. For consistency sake, I have used only one button style (primary) for all buttons on Kedro-viz. Additionally added the size small since it was used in the experiment tracking. I have removed secondary button style (which was also used for experiment tracking) and also removed the wipe transition (used in large pipeline warning) for the sake of consistency and simplicity. 

UPDATE - As per the discussion with Tynan, and Gabriel in the check-in, I have added secondary mode for buttons (since it's used). I have removed wipe transition which was used in large pipelines - and Gabriel is comfortable with it. 
I have also removed disable since we are not using it atm. 

## Checklist

- [ ] Read the [contributing](/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added new entries to the `RELEASE.md` file
- [x] Added tests to cover my changes

### CODE REVIEW

**Comment by tynandebold** [comment]
Just a quick comment off the top: I think we need to keep the secondary button style (and really any other styles that were used in Viz).

For example, the edit runs modal has two button styles:
Screen Shot 2022-01-25 at 10 03 44

Though with your change it looks like this:
Screen Shot 2022-01-25 at 10 02 19

There should be some differentiation between those two.

File: src/components/experiment-wrapper/experiment-wrapper.js [hunk 1]
```diff
@@ -181,7 +181,7 @@ const ExperimentWrapper = ({ theme }) => {
             rel="noreferrer"
             target="_blank"
           >
-            <Button theme={theme}>View docs</Button>
```
Comment (tynandebold) [inline src/components/experiment-wrapper/experiment-wrapper.js:184]: I noticed you're removing the `theme` prop here, which is fine, but didn't remove it on the buttons in the `src/components/export-modal/export-modal.js` file. I wonder why? I think you can remove it there as well can't you?

**Review by tynandebold — APPROVED** [review]
Looks great! Just one tiny comment about not removing the `theme` prop in one file, but it's not a blocker :)

File: src/components/experiment-wrapper/experiment-wrapper.js (hunk 1 above)
Comment (rashidakanchwala) [reply src/components/experiment-wrapper/experiment-wrapper.js:184]: yes, i will remove it :) .. thanks!

File: src/components/button/button.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+/**
+ * Generic Kedro Button
+ */
+const Button = ({ children, onClick, size, mode }) => (
+  <span className="kedro kui-button">
+    <button
+      className={classnames(
+        'kui-button__btn',
+        `kui-button__btn--${size}`,
+        `kui-button__btn--${mode}`
```
Comment (AntonyMilneQB) [inline src/components/button/button.js:15]: Just curious... what's up with the ` instead of '?

Comment (rashidakanchwala) [reply src/components/button/button.js:15]: It's called template literals similar to f-string or % in python ( https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals)

Comment (AntonyMilneQB) [reply src/components/button/button.js:15]: Ahh cool, thank you!

**Review by studioswong — COMMENTED** [review]
I just had an initial skim through and tested the button, looks good! I'll do a more through dive into the code and let you know if I spot anything.

Please update the release notes for this feature in the meantime :)

File: src/components/button/button.scss [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+.kui-theme--dark {
+  --color-button--active: #{rgba($color-light, 0.85)};
+  --color-button--outline: #{$color-light};
+  --color-button__text--hover: #{rgba($color-dark, 0.85)};
+  --color-button__text: #{rgba($color-light, 0.85)};
+}
+
+.kui-button {
```
Comment (studioswong) [inline src/components/button/button.scss:28]: Please remove references of `kui` and rename all css classnames for this component to either `kedro-button` or just `button`.

**Review by studioswong — CHANGES_REQUESTED** [review]
The code overall looks good! Requesting changes mainly for renaming the button classes as we are moving away from kui references. (I understand we have used `kui-theme` for the theme naming for the overall app given our previous setup and links to kui, which I have put in a ticket to rename those class name to strip away the kui reference( just `theme--light` or `theme--dark` ) and can be done in one go after we have migrated all kui elements.)

**Review by studioswong — APPROVED** [review]
Nice one 👏

**Pull request merged**